### ISSUE TITLE
Use local Onyx key to throttle location permission prompts

### ISSUE THREAD

**Issue by garrettmknight** [Expensify/App#47800]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** N/A
**Reproducible in staging?:** N/A
**Reproducible in production?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:** https://github.com/Expensify/Expensify/issues/378927
**Issue reported by:** @Julesssss 
**Slack conversation:**

## Action Performed:
1. Sign up with a new account
2. Big + > Submit expense
3. Scan receipt
4. Select a recipient
5. See the location services request > tap Not Now
6. When the system request comes up, tap 'Don't Allow'
7. Big + > Submit expense
8. Scan receipt
9. Select a recipient

## Current Result:
Every time the user creates a money request, we prompt them to enable the location permission.

## Requested Result:
Instead of waiting for a backend solution, we can use a local Onyx key/timestamp to prevent the modal from showing. Let's implement logic that adds a one-week delay between each prompt of the location permission.

## Workaround:
N/A

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

![image](https://github.com/user-attachments/assets/1468199a-0cb2-443c-9373-500780ac0065)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by slafortune** [comment]
Updated the steps to include submitting the receipt which needs to be done before the location services request populates.

**Comment by trjExpensify** [comment]
@shubham1206agra what do you think about these proposals?

**Comment by Julesssss** [comment]
I agree the timestamp is enough here. Lets figure out the rest of the minor details in the PR review.

**Comment by Julesssss** [comment]
Reassigning Tim as I didn't see his 'ssigned Krishna2323 13 hours ago' action

**Comment by trjExpensify** [comment]
>Disable the location check for iOS users on the quick flow action (this would include users who did grant permission)

This sounds non-ideal, because the quick action is likely to be used repeatedly as the preferred method to submit expenses to your company over and over again. If I'm understanding you, disabling it would mean the added accuracy upside of location checking would not apply to this core flow. 

>iOS users who approve our modal, but select 'Allow Once' on the native iOS prompt will continue to be prompted when using the quick action flow. 

Won't they also be prompted on every `Submit expense` attempt as well? That's kinda' the nature of choosing "Allow once" isn't it, or am I missing something on why this only applies to the quick action flow?

**Comment by Julesssss** [comment]
> disabling it would mean the added accuracy upside of location checking would not apply to this core flow.

Yeah, exactly. My thinking is that this is slightly preferrable to the frustrating native location prompt on every single request. 

> Won't they also be prompted on every Submit expense attempt as well?

In the submit expense flow our 7-day throttling will prevent them from seeing the prompt every time. Additionally we are able to show our custom prompt ahead of the native-iOS prompt -- which is slightly less annoying as they at least are given context for the prompt.

**Comment by trjExpensify** [comment]
Why aren't we applying the same logic to the QAB as we are `Submit expense`?

**Comment by trjExpensify** [comment]
>My thinking is that this is slightly preferrable to the frustrating native location prompt on every single request.

That's your own doing though, isn't it. You selected `Only once`, not `Always`. If it's annoying you, change it to `Always`.

**Comment by Julesssss** [comment]
Fair point 😄 

It makes the whole 'provide context around this contextless location permission' feature redundant. If users didn't select 'always' after going through our context flow I would suggest they are more likely to select 'never' when given the no context apple prompt. Though as it just occurs for iOS quick action IF users selected not now I suppose this is the better option.

**Comment by slafortune** [comment]
not deployed yet

**Comment by Julesssss** [comment]
PR has had comments addressed and oday I requested an additional test for the prompt to reshow.

**Comment by Julesssss** [comment]
Merged!

**Comment by slafortune** [comment]
So that I can process these payment - @shubham1206agra  Please propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the new feature will work correctly on production in further releases.

**Comment by Julesssss** [comment]
Waiting on regression test

**Comment by slafortune** [comment]
@shubham1206agra looks like the original offer is expired - please accept the new offer I sent [here](https://www.upwork.com/nx/wm/offer/104518555)

**Comment by slafortune** [comment]
@Krishna2323Paid  ✅

**Comment by slafortune** [comment]
@shubham1206agra Paid ✅
